atomics/include/desul/atomics/Adapt_SYCL.hpp

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright (c) 2019, Lawrence Livermore National Security, LLC
+and DESUL project contributors. See the COPYRIGHT file for details.
+Source: https://github.com/desul/desul
+
+SPDX-License-Identifier: (BSD-3-Clause)
+*/
+
+#ifndef DESUL_ATOMICS_ADAPT_SYCL_HPP_
+#define DESUL_ATOMICS_ADAPT_SYCL_HPP_
+
+#include <desul/atomics/Common.hpp>
+
+// FIXME_SYCL SYCL2020 dictates that <sycl/sycl.hpp> is the header to include
+// but icpx 2022.1.0 and earlier versions only provide <CL/sycl.hpp>
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+namespace desul {
+namespace Impl {
+
+#ifdef __clang__
+namespace sycl_sync_and_atomics = ::sycl::ext::oneapi;
+#else
+namespace sycl_sync_and_atomics = ::sycl;
+#endif
```

Comment:
We should likely try doing something less brittle. It should be fine for this pull request, but we should put a `FIXME` here linking to https://github.com/desul/desul/pull/21#issuecomment-877468175.

It also depends on the compiler versions we want to support in `desul`, of course.

Reply:
> We should likely try doing something less brittle. 

I am not sure what you mean.  This is "unchanged" just being moved around.  Are you saying that we should revisit that in an upcoming PR?

> but we should put a FIXME here linking to https://github.com/desul/desul/pull/21#issuecomment-877468175.

Sure, will add.

Reply:
> I am not sure what you mean. This is "unchanged" just being moved around. Are you saying that we should revisit that in an upcoming PR?

Yes, I'm aware that this wasn't changed and we should revisit it again (after deciding which compiler and versions to support).